### PR TITLE
session: make unit test TestOptimisticConflicts more stable

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/12410

### What is changed and how it works?

A simplified order looks like this:

```
begin pessimistic
wait for a long time
truncate table
```

The DDL operation needs to commit the transaction first.
This commit operation failed because of the pessimistic lock has gone.

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood @SunRunAway @wjhuang2016

File: store/tikv/snapshot_test.go [hunk 1]
```diff
@@ -211,9 +211,11 @@ func (s *testSnapshotSuite) TestLockNotFoundPrint(c *C) {
 }
 
 func (s *testSnapshotSuite) TestSkipLargeTxnLock(c *C) {
+	x := kv.Key("x_this_one_run_parallel_with_others?")
```
Comment (shenli) [inline store/tikv/snapshot_test.go]: s/"x_this_one_run_parallel_with_others"/"x_key_TestSkipLargeTxnLock"/

Comment (tiancaiamao) [reply store/tikv/snapshot_test.go]: Done

**Pull request merged**